storage-proofs/src/stacked/params.rs

```diff
@@ -181,20 +176,11 @@ impl<H: Hasher, G: Hasher> Proof<H, G> {
         check!(challenge < graph.size());
         check!(pub_inputs.tau.is_some());
 
-        // just grabbing the first one
-        let actual_comm_r = self.comm_r();
-        let expected_comm_r = if let Some(ref tau) = pub_inputs.tau {
-            &tau.comm_r
-        } else {
-            return false;
-        };
-
-        check_eq!(expected_comm_r, &actual_comm_r);
-
```

Comment:
Good that you no longer recalculate `comm_r`, but I think you still need to verify that that any `comm_c` or `comm_r_last` used in the individual proofs are actually the same as the ones used to verify `comm_r` once at the beginning.